docs/build/js-client/4-advanced-ipfs.md

```diff
@@ -0,0 +1,297 @@
+# Advanced example
+
+## Intro
+
+In this section, we will demonstrate how the Fluence JS Client can be used to create a IPFS client application with the Fluence stack.
+
+We will implement IPFS interface for interacting with IPFS p2p network.
+
+## Aqua code
+
+Let's start with the Aqua code first:
+
+```aqua
+alias CID : string                                  -- (1)
+
+service IpfsClient("ipfs client"):                  -- (2)
```

Comment:
Why not import it from https://github.com/fluencelabs/aqua-ipfs/tree/master/aqua?